app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterClaimController.scala

```diff
@@ -152,28 +158,49 @@ class EnterClaimController @Inject() (
       }
     }
 
-  def checkClaim(): Action[AnyContent] =
+  def checkClaimSummary(): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
       withAnswers[ClaimsAnswer] { (_, answers) =>
         answers match {
-          case Some(claims) => Ok(checkClaimPage(claims))
+          case Some(claims) => Ok(checkClaimSummaryPage(claims, checkClaimAnswerForm))
           case None         => Redirect(routes.EnterClaimController.startClaim())
         }
       }
     }
 
-  def checkClaimSubmit(): Action[AnyContent] =
+  def checkClaimSummarySubmit(): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
-      withAnswers[ClaimsAnswer] { (fillingOutClaim, _) =>
-        fillingOutClaim.draftClaim
-          .fold(_.movementReferenceNumber)
-          .fold {
-            logger.warn("could not find movement reference number")
-            errorHandler.errorResult()
-          }(referenceNumber =>
-            referenceNumber.value match {
-              case Left(_)  => Redirect(routes.BankAccountController.enterBankAccountDetails())
-              case Right(_) => Redirect(routes.BankAccountController.checkBankAccountDetails())
+      withAnswers[ClaimsAnswer] { (fillingOutClaim: FillingOutClaim, _) =>
+        EnterClaimController.checkClaimAnswerForm
+          .bindFromRequest()
+          .fold(
+            formWithErrors =>
+              fillingOutClaim.draftClaim
+                .fold(_.claimsAnswer)
+                .map(claims => Future.successful(BadRequest(checkClaimSummaryPage(claims, formWithErrors))))
+                .getOrElse(Future.successful(errorHandler.errorResult())),
+            { claims =>
+              val newDraftClaim  = fillingOutClaim.draftClaim.fold(_.copy(checkClaimAnswer = Some(claims)))
+              val updatedJourney = fillingOutClaim.copy(draftClaim = newDraftClaim)
+
+              val result = EitherT(updateSession(sessionStore, request)(_.copy(journeyStatus = Some(updatedJourney))))
+                .leftMap(_ => Error("could not update session"))
+
+              result.fold(
+                e => {
+                  logger.warn("could not get claims details", e)
```

Comment:
you're not getting any claims details here - you're really just storing the radio button answer